### ISSUE TITLE
Missing refresh tokens for Resource Owner flow in CAS 5.1.0

### ISSUE THREAD

**Issue by klewczuk** [apereo/cas#2731]
Hi,

When using Resource Owner grant in CAS 5.1.0 I get an access token, but no refresh token is generated. I get the following response:

```json
{
"access_token": "AT-1-0HzePyf5n5qq3fE1wVuDZJdexVJJLsy7nhf",
"token_type": "bearer",
"expires_in": 7200
}
```

Of course I have defined my service to generate refresh tokens (I've set "generateRefreshToken": "true" in service json file).
In CAS 5.0.4 there was a refresh token generated together with access token. Was it removed intentionally in CAS 5.1.0? Or it is a bug?

**Comment by mmoayyed** [comment]
Sounds like a bug to me. Do your logs reveal anything useful?

**Comment by klewczuk** [comment]
Hi,

I've found nothing interesting in logs, but after debugging I see that for Resource Owner an AccessTokenRequestDataHolder is used with the constructor that always sets generateRefreshToken flag to false. So no matter if service json file defines to generate refresh token or not, we will never get it. Here is a shortcut of the stacktrace:

```
Daemon Thread [https-jsse-nio-8443-exec-5] (Suspended (breakpoint at line 51 in AccessTokenRequestDataHolder))		
	AccessTokenRequestDataHolder.<init>(Service, Authentication, OAuthRegisteredService, TicketGrantingTicket) line: 51	
	AccessTokenPasswordGrantRequestExtractor.extract() line: 79	
	OAuth20AccessTokenEndpointController.examineAndExtractAccessTokenGrantRequest(HttpServletRequest, HttpServletResponse) line: 189	
	OAuth20AccessTokenEndpointController.handleRequest(HttpServletRequest, HttpServletResponse) line: 111
```

Later in:
```OAuth20AccessTokenEndpointController.handleRequest(HttpServletRequest, HttpServletResponse) line: 124```
there is a check made:
```
if (responseHolder.isGenerateRefreshToken()) {
    refreshToken = generateRefreshToken(responseHolder);
    LOGGER.debug("Refresh Token: [{}]", refreshToken);
} else {
    LOGGER.debug("Service [{}] is not able/allowed to receive refresh tokens", responseHolder.getService());
}
```
which will always return false and skip refresh token generation.

When using Auhtorization Code grant there is other extractor used (AccessTokenAuthorizationCodeGrantRequestExtractor), which gets the service and checks the refresh token generation flag:
`AccessTokenAuthorizationCodeGrantRequestExtractor.extract() line: 47 & line:51`
so I guess AccessTokenPasswordGrantRequestExtractor should do something similar to have refresh token generation working properly.

**Comment by mmoayyed** [comment]
You should be able to test the fix in about an hour or so, by using the new SNAPSHOT.

**Comment by klewczuk** [comment]
Thanks for fixing it.

**Comment by klewczuk** [comment]
@mmoayyed Could this fix be merged to 5.1.x branch and released in 5.1.2? This is critical bug for me and I would like to have it fixed in 5.1.2 rather than waiting for 5.2.0 in November.
